action.yml

```diff
@@ -24,5 +28,30 @@ inputs:
     default: 'gpt-4-turbo-preview'
 
 runs:
-  using: 'docker'
-  image: 'Dockerfile' 
\ No newline at end of file
+  using: 'composite'
+  steps:
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.12'
+        cache: 'pip'
+    
+    - name: Install dependencies
```

Comment:
🔒 Make sure to validate the environment variables before running the Python script to avoid runtime errors. You can add checks to ensure that required variables are set.